accounts/abi/bind/backends/simulated.go

```diff
@@ -354,6 +354,14 @@ func (b *SimulatedBackend) CallContract(ctx context.Context, call ethereum.CallM
 	if err != nil {
 		return nil, err
 	}
+	// If the result contains a revert reason, unpack and return it.
+	if res.Err != nil && len(res.Revert()) > 0 {
+		reason, err := abi.UnpackRevert(res.Revert())
+		if err != nil {
```

Comment:
If the error decoding is failed, shouldn't we return the raw error message here? Otherwise, the returned error is for abi decoding.

Reply:
Yep, changed it, thanks!